api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/aup/AcceptableUsagePolicyProperties.java

```diff
@@ -80,6 +80,30 @@ public MongoDb() {
          * The table name in the database that holds the AUP attribute to update for the user.
          */
         private String tableName;
+        
+        /**
+         * The column to store the AUP attribute. May differ from the profile attribute defined by <code>aupAttributeName</code>.
+         * SQL query can be further customized by setting <code>sqlUpdateAUP</code>.
+         */
+        private String aupColumn;
+        
+        /**
+         * The column to idetify the principal.
+         * SQL query can be further customized by setting <code>sqlUpdateAUP</code>.
+         */
+        private String principalIdColumn = "username";
+        
+        /**
+         * The profile attribute to extract the value for the <code>principalIdColumn</code> used in the WHERE clause
```

Comment:
Consider using Javadoc linking tags such as `{@link}` rather than `code` tags. They work better in the future when things get removed or refactored and the javadoc building process can catch those errors. Do this for all docs for the settings you have added.

Reply:
Okay